cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2866,12 +2884,18 @@ String getInternalServiceHostname(String serviceName)    {
         }
 
         Future<ReconciliationState> kafkaExporterSecret(Supplier<Date> dateSupplier) {
-            return secretOperations.reconcile(namespace, KafkaExporter.secretName(name),
-                    kafkaExporter == null ? null : kafkaExporter.generateSecret(clusterCa, isMaintenanceTimeWindowsSatisfied(dateSupplier)))
-                    .map(res -> {
-                        isKafkaExporterCertsChanged = res instanceof ReconcileResult.Patched;
-                        return this;
-                    });
+            return secretOperations.getAsync(namespace, KafkaExporter.secretName(name))
+                    .compose(oldSecret -> secretOperations.reconcile(namespace, KafkaExporter.secretName(name),
+                            kafkaExporter == null ? null : kafkaExporter.generateSecret(clusterCa, isMaintenanceTimeWindowsSatisfied(dateSupplier)))
+                            .map(res -> {
+                                if (res instanceof ReconcileResult.Patched) {
+                                    // The secret is patched and some changes to the existing certificates actually occured
+                                    isExistingKafkaExporterCertsChanged = ModelUtils.didAnyCertificateChangedInSecret(oldSecret, res.resource());
+                                }
+
+                                return this;
+                            })
+                    );
         }
```

Comment:
We've got this same pattern of method calls repeated three or four times in this PR. Surely we can factor out a common method which takes the secret name and `BiFunction` whose argument is a method reference to `kafkaExporter::generateSecret` or whatever?

Reply:
I migth have done this a bit differently, but I factored it out. and re-used it in all 4 methods.